test/functional/api/v2/hostgroups_controller_test.rb

```diff
@@ -63,6 +63,32 @@ class HostgroupsControllerTest < ActionController::TestCase
         assert_response :success
         assert_equal assigns('hostgroup').ansible_roles, [@ansible_role2, @ansible_role1]
       end
+
+      test 'should append a role to a hostgroup with a correct ordering' do
+        hostgroup = FactoryBot.create(:hostgroup,
+                                      :ansible_role_ids => [@ansible_role1.id])
+        post :ansible_roles,
+             :params => {
+               :id => hostgroup.id,
+               :ansible_role_id => [@ansible_role2.id]
+             },
+             :session => set_session_user
+        assert_response :success
+        assert_equal assigns('hostgroup').ansible_roles, [@ansible_role1, @ansible_role2]
+      end
+
+      test 'should remove only specified roles from a hostgroup' do
```

Comment:
Another use case: what if the role doesn't exist? Does it return 404 or 200/204 since it's not part of the role collection anymore? I'd expect the latter.

Reply:
The server will respond with `200`.